tests/test_gruptree.py

```diff
@@ -217,22 +281,61 @@ def test_grupnetroot(schstr, expected_dframe, expected_tree):
         expected_dframe.sort_values(["CHILD", "PARENT"]).reset_index(drop=True),
         check_dtype=False,
     )
-    gruptreedict = gruptree.edge_dataframe2dict(grupdf)
-    treelibtree = gruptree.dict2treelib("", gruptreedict[0])
-    assert str(treelibtree).strip() == expected_tree.strip()
+    treelist = gruptree.edge_dataframe2dict(grupdf)
+    # Merge strings for all trees (if multiple roots)
+    strtrees = [str(gruptree.dict2treelib(tree)) for tree in treelist]
+    strtrees.sort()  # Avoid flaky test due to sorting
+    treelibtree = "".join(strtrees)
+    assert treelibtree.strip() == expected_tree.strip()
+
+
+def test_multiple_roots():
+    """Test edge_dataframe2dict with multiple roots"""
+    edges = pd.DataFrame(
+        [
+            {"CHILD": "FIELDA", "PARENT": None},
+            {"CHILD": "FIELDB", "PARENT": None},
+            {"CHILD": "PLATA", "PARENT": "FIELDA"},
+            {"CHILD": "PLATB", "PARENT": "FIELDB"},
+        ]
+    )
+    list_of_treedicts = gruptree.edge_dataframe2dict(edges)
+    assert len(list_of_treedicts) == 2
+    assert {"FIELDA": {"PLATA": {}}} in list_of_treedicts
+    assert {"FIELDB": {"PLATB": {}}} in list_of_treedicts
```

Comment:
Could you not simply assert the `list_of_treedicts` in a single go? Or does the `gruptree.edge_dataframe2dict` not guarantee order? Would also be possible to make it more obvious that the same result is expected twice - and reduce overhead.

Reply:
Fixed.

Reply:
Fixed once more to guarantee sorting by root-node-name.